### PR TITLE
Support video demand requests for video-enabled adUnits

### PR DESCRIPTION
**Pull request by matthewlane**

### CODE REVIEW

File: adapters.json [hunk 1]
```diff
@@ -34,5 +34,10 @@
       "appnexus": {
         "alias": "pagescience"
       }
+    },
+    {
+      "appnexusAst": {
+        "supportedMediaTypes": ["video"]
+      }
```
Comment (mkendall07) [inline adapters.json]: @protonate 
What are feelings on this pattern? Supported types specified at the configuration level...

File: src/prebid.js [hunk 2]
```diff
@@ -272,6 +272,10 @@ function removeComplete() {
     .forEach(bid => responses.slice(responses.indexOf(bid), 1));
 }
 
+const videoAdUnit = adUnit => adUnit.mediaType === 'video';
+const nonVideoBidder = bid => !adaptermanager.videoAdapters.includes(bid.bidder);
+const hasNonVideoBidder = adUnit => adUnit.bids.filter(nonVideoBidder).length;
+
```
Comment (mkendall07) [inline src/prebid.js]: These lines feel out of place here. Maybe create a video.js file and put them in there instead?

**Comment by mkendall07** [comment]
LGTM after review notes addressed. Thanks!

File: adapters.json (hunk 1 above)
Comment (protonate) [inline adapters.json]: It seems an adapter would know it's supported types, unless that would be an optional config for a publisher to choose. The `adaptermanager._bidderRegistry` could be queried by a function exported by `adaptermanager` to get the array of `videoAdapters`.

**Pull request merged**